src/cascadia/TerminalSettingsModel/Profile.cpp

```diff
@@ -266,11 +266,23 @@ winrt::hstring Profile::ExpandedIconPath() const
 // - This profile's expanded background image path / the empty string.
 winrt::hstring Profile::ExpandedBackgroundImagePath() const
 {
+    WCHAR desktopWallpaper[MAX_PATH];
     if (_BackgroundImagePath.empty())
     {
         return _BackgroundImagePath;
     }
-    return winrt::hstring{ wil::ExpandEnvironmentStringsW<std::wstring>(_BackgroundImagePath.c_str()) };
+    // checks if the user would like to copy their desktop wallpaper
+    // if so, replaces the path with the desktop wallpaper's path
+    else if (_BackgroundImagePath == to_hstring("DesktopWallpaper"))

```

Comment:
(regarding the static string) At the top of this file, we have a bunch of these:
```
static constexpr std::string_view NameKey{ "name" };
```

Just add a new one called `DesktopWallpaperKey` and reference it here. Though this isn't a json key, so maybe name it `DesktopWallpaperEnum` instead? Up to you.

For some background, all of those static strings are what we look for in our JSON. You'll find some of these in just about every settings object cpp file. As we add more strings like this, makes sense to throw them in there too :)

Reply:
Got it. This makes way more sense.